extensions/markdown-math/src/extension.ts

```diff
@@ -19,24 +21,34 @@ export function activate(context: vscode.ExtensionContext) {
 		return config.get<{ [key: string]: string }>('math.macros', {});
 	}
 
+	function preprocessMarkdown(markdown: string): string {
+		const fencedMathRegex = /```math\s+([\s\S]*?)\s+```/g;
```

Comment:
I don't think a regular expression is the right approach. It's fragile and won't handle a lot of cases without extra work (a fetched code block can use more than 3 ``` ` ``` for example, or use `~~~`)

Instead can we override the fenced code block parser to handle the `math` language specially?

Reply:
Yea I can do this. Just to be clear we want to move the changes back to markdown-language-features, adding a type of fenced code block for math blocks?